### PR TITLE
cgroup: Optionally add process and task to a subsystems subset

### PR DESCRIPTION
**Pull request by sameo**
In some cases, one may want to add processes and task only to certain
subsystems in a cgroup. For example, when a cgroup is created to only
limit a given subsystem usage, but leaves all other subsystems
unconstrained, there may be a need to move a process to only the
contrained cgroup subsystem while leaving it in another cgroup for
other subsystems.

This commit makes the Control interface process and task addition
functions variadic, in order to pass an optional list of subsystems that
behave as s subsystems filter.

Signed-off-by: Samuel Ortiz <samuel.e.ortiz@protonmail.com>

### CODE REVIEW

**Comment by fuweid** [comment]
Need to rebase

**Comment by sameo** [comment]
@fuweid The PR is rebased now.

**Review by mxpv — APPROVED** [review]

**Review by kzys — APPROVED** [review]

File: cgroup.go [hunk 1]
```diff
@@ -149,22 +149,40 @@ func (c *cgroup) Subsystems() []Subsystem {
...
+		}
+	}
+
+	return filteredSubsystems
+}
+
 // Add moves the provided process into the new cgroup
-func (c *cgroup) Add(process Process) error {
-	return c.add(process, cgroupProcs)
+func (c *cgroup) Add(process Process, subsystems ...Name) error {
```
Comment (mikebrow) [inline cgroup.go:171]: Can we update the inline function descriptions too..

File: control.go [hunk 2]
```diff
@@ -54,12 +54,22 @@ type Task = Process
...
+	// the process is added to all the cgroup subsystems. When giving Add a list of
+	// subsystem names, the process is only added to those subsystems, provided that
+	// they are active in the targeted cgroup.
+	Add(Process, ...Name) error
+	// AddProc adds the process with the given id to the cgroup (cgroup.procs).
+	// Without additional arguments, the process with the given id is added to all
+	// the cgroup subsystems. When giving AddProc a list of subsystem names, the process
+	// id is only added to those subsystems, provided that they are active in the targeted
+	// cgroup.
+	AddProc(pid uint64, subsystems ...Name) error
```
Comment (mikebrow) [inline control.go:67]: declaration isn't consistent with the other two.. as it includes subsystems

File: cgroup_test.go [hunk 3]
```diff
@@ -101,6 +101,50 @@ func TestAdd(t *testing.T) {
 	}
 }
 
+func TestAddFilteredSubsystems(t *testing.T) {
```
Comment (mikebrow) [inline cgroup_test.go:104]: let's also test against nil filteredSubstems..

**Review by mikebrow — COMMENTED** [review]
see comments..

Does Update need to be have filters too?
